services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/FlowService.java

```diff
@@ -0,0 +1,213 @@
+     */
+    public void deleteFlow(String flowId) {
+        TransactionManager transactionManager = persistenceManager.getTransactionManager();
+        transactionManager.begin();
+        try {
+            Iterable<Flow> flows = flowRepository.findById(flowId);
+            for (Flow flow : flows) {
+                processDeleteFlow(flow);
+            }
+            transactionManager.commit();
+        } catch (Exception e) {
+            transactionManager.rollback();
+            throw e;
+        }
+    }
+
+    /**
+     * Get FlowPair by the FlowId.
+     * @param flowId - flow identificator
+     * @return FlowPair object
+     */
+    public FlowPair getFlowPair(String flowId) {
+        Iterable<Flow> flows = flowRepository.findById(flowId);
+        FlowPair.FlowPairBuilder flowPairBuilder = FlowPair.builder();
+        for (Flow flow: flows) {
+            if (flow.isForward()) {
+                flowPairBuilder.forward(flow);
+            } else {
+                flowPairBuilder.reverse(flow);
+            }
+        }
+        return flowPairBuilder.build();
+    }
+
+    /**
+     * Get all flows grouped in FlowPairs.
+     * @return List of flow pairs
+     */
+    public List<FlowPair> getFlows() {
+        Collection<Flow> flows = flowRepository.findAll();
+        Map<String, FlowPairBuilder> flowPairs = new HashMap<>();
+        for (Flow f : flows) {
+            String flowId = f.getFlowId();
+            FlowPair.FlowPairBuilder flowPairBuilder;
+            if (flowPairs.containsKey(flowId)) {
+                flowPairBuilder = flowPairs.get(flowId);
+            } else {
+                flowPairBuilder = FlowPair.builder();
+                flowPairs.put(flowId, flowPairBuilder);
+            }
+            if (f.isForward()) {
+                flowPairBuilder.forward(f);
+            } else {
+                flowPairBuilder.reverse(f);
+            }
+
+        }
+        List<FlowPair> flowPairList = new ArrayList<>();
+        for (FlowPairBuilder builder :flowPairs.values()) {
+            flowPairList.add(builder.build());
+        }
+        return flowPairList;
+    }
+
+    /**
+     * Update status for selected flow.
+     * @param flowId - target flow to update
+     * @param flowStatus - new status
+     * @return target FlowPair
+     */
+    public FlowPair updateFlowStatus(String flowId, FlowStatus flowStatus) {
+        TransactionManager transactionManager = this.persistenceManager.getTransactionManager();
+        transactionManager.begin();
+        try {
+            FlowPair pair = getFlowPair(flowId);
+            Flow forward = pair.getForward();
+            Flow reverse = pair.getReverse();
+            if (forward != null && reverse != null) {
+                forward.setStatus(flowStatus);
+                flowRepository.createOrUpdate(forward);
+
+                reverse.setStatus(flowStatus);
+                flowRepository.createOrUpdate(reverse);
+            }
```

Comment:
there is the "else" branch with exception?

Reply:
Refactored.